src/main/java/org/folio/service/RequestExpirationService.java

```diff
@@ -246,4 +246,29 @@ private Future<RowSet<Row>> resetPositionsForOpenRequests(Conn conn,
   private Future<Void> updateRequest(Conn conn, Request request) {
     return conn.update(REQUEST_TABLE, request, request.getId()).mapEmpty();
   }
+
+  private Void publishPubSubLogEvents(List<ExpiredRequestWrapper> context) {
+    context.forEach(requestWrapper -> {
+      var payload = new JsonObject()
+        .put(REQUESTS.value(), new JsonObject()
+          .put(ORIGINAL.value(), requestWrapper.originalValue())
+          .put(UPDATED.value(), requestWrapper.updatedValue()));
+      eventPublisherService.publishLogRecord(payload, REQUEST_EXPIRED);
+    });
+
+    return null;
+  }
+
+  private Future<Void> publishRequestDomainEvents(List<ExpiredRequestWrapper> context) {
+    Future<Void> future = succeededFuture();
+    for (ExpiredRequestWrapper requestWrapper : context) {
+      Request oldEntity = requestWrapper.originalValue().mapTo(Request.class);
+      Request newEntity = requestWrapper.updatedValue().mapTo(Request.class);
+      future = future.compose(v ->
```

Comment:
This way one failed event publishing will fail the whole process, which means all subsequent events in the list will never be published. Is that what we want?